hyperspy/learn/mva.py

```diff
@@ -1317,22 +1369,666 @@ def _estimate_elbow_position(self, curve_values):
+            each measurement is scaled to length 1. 
+            You can also pass one of the scikit-learn preprocessing 
+            scale_method = import sklearn.processing.StandadScaler
+            scaling = scale_method
+            See scaling methods in scikit-learn preprocessing for further 
+            details.
+        scaling_kwargs : dict, default empty
+            Additional parameters passed to the cluster scaling algorithm. 
+            See sklearn.preprocessing scaling methods for further details
+        use_decomposition_results : bool, default : True
+            If True (recommended) the signal's decomposition results are used
+            for clustering. Note - If this option is not used the raw data
+            is used. This can be memory intensive and is only recommened if
+            the Signal has a small `signal_dimension`.
+        use_decomposition_for_centers : bool
+            If True (recommended) the pca results are used for building the 
+            the cluster centers from the clustered label results. 
+            If False the original signal data is used.
+        number_pca_components : int, default None
+            If you are getting the cluster centers using the decomposition 
+            results (use_decomposition_for_centers=True) you can define how 
+            many PCA components to use. If set to None the method uses the 
+            estimate of significant components found in the decomposition step
+            using the elbow method and stored in the 
+            ``learning_results.number_significant_components`` attribute.
+        navigation_mask : boolean numpy array, default : None
+            The navigation locations marked as True are not used in the
+            decomposition.
+        signal_mask : boolean numpy array, default : None
+            The signal locations marked as True are not used in the
+            decomposition.
+        metric : {'elbow','silhouette','gap'} default 'gap'
+            Use distance,silhouette analysis or gap statistics to estimate 
+            the optimal number of clusters.
+            Gap is believed to be, overall, the best metric but it's also 
+            the slowest. Elbow measures the distances between points in 
+            each cluster as an estimate of how well grouped they are and 
+            is the fastest metric. 
+            For elbow the optimal k is the knee or elbow point. 
+            For gap the optimal k is the first k gap(k)>= gap(k+1)-std_error
+            For silhouette the optimal k will be one of the "maxima" found with
+            this method
+        n_ref :  int, default 5
+            Number of random references to use in gap statistics method
+            Gap statistics compares the results from clustering the data to
+            clustering random data. This random clustering is
+            typically averaged n_ref times to get an statistical average
+        cluster_kwargs: dict {}  default empty
```

Comment:
change to `**kwargs` to match the method definition.

Reply:
fixed